lib/runtime/scheduler.js

```diff
@@ -1,10 +1,14 @@
 var Heap = require('heap');
 var Base = require('extendable-base');
+var errors = require('../errors');
 
 var Scheduler = Base.extend({
     initialize: function(options) {
         this.timer = null;
         this.queue = new Heap(function(a, b) { return a.ms - b.ms; });
+
+        this.running = false;
+        this.stopped = false;
```

Comment:
It's not immediately obvious why you have two booleans that at first glance imply the same thing -- upon reading the code it's clear that the thing you're trying to effect is that someone can schedule before start is called but not after stop is called, i.e. there is a difference between `running` and `! stopped`.

Perhaps this would be clearer if you used a `state` variable that could be one of `INIT`, `RUNNING` or `STOPPED`?